### PR TITLE
Added Junit Tests for LocalCache Class

### PR DESCRIPTION
**Pull request by sonja-ct**
Added Junit Tests for LocalCache Class

Signed-off-by: Sonja <sonja.matic@comtrade.com>

**Related Issue**
/
**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
hey @Coduz,
I have checked this PR and it looks OK - can you please check it when you have time and merge?

File: commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+        int invalidNegativeSizeMax = -1;
+        int invalidNegativeExpireAfter = -1;
+        IllegalArgumentException illegalArgumentExceptionSizeMax = new IllegalArgumentException("maximum size must not be negative");
+        IllegalArgumentException illegalArgumentExceptionExpireAfter = new IllegalArgumentException("duration cannot be negative: " + invalidNegativeExpireAfter + " SECONDS");
+
+        for (int sizeMax : sizeMaxList) {
+            for (int expireAfter : expireAfterList) {
+                for (Object defaultValue : defaultValueList) {
+                    LocalCache<Object, Object> localCache = new LocalCache<>(sizeMax, expireAfter, defaultValue);
+                    assertNotNull(localCache);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+                    assertEquals("Expected and actual values should be the same.", defaultValue, localCache.get(defaultValueList));
+                    assertThat("LocalCache object expected.", localCache, IsInstanceOf.instanceOf(LocalCache.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+    public void localCacheTest2() {
+        int[] sizeMaxList = new int[]{0, 10, 1000, 10000, 2147483647};
+        Object[] defaultValueList = new Object[]{0, 10, 100000, "String", 'c', -10, -1000000000, -100000000000L, 10L, 10.0f, null, 10.10d, true, false,};
+        int invalidNegativeSizeMax = -1;
+        IllegalArgumentException illegalArgumentExceptionSizeMax = new IllegalArgumentException("maximum size must not be negative");
+
+        for (int sizeMax : sizeMaxList) {
+            for (Object defaultValue : defaultValueList) {
+                LocalCache<Object, Object> localCache = new LocalCache<>(sizeMax, defaultValue);
+                assertNotNull("Null not expected.", localCache);
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

File: commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+                assertEquals("Expected and actual values should be the same.", defaultValue, localCache.get(defaultValueList));
+                assertThat("LocalCache object expected.", localCache, IsInstanceOf.instanceOf(LocalCache.class));
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/cache/LocalCacheTest.java]: Same things as #3008 - https://github.com/eclipse/kapua/pull/3008/files#diff-031d59073d14c177c331d7ce9d2a83d8R67

**Review by Coduz — CHANGES_REQUESTED** [review]
I made some comments on things which I have already commented on #3008.
All other things are good :)

**Comment by sonja-ct** [comment]
Hey @Coduz,
I made all the requested changes, if you can take a look, I hope everything is ok now.
Thanks, Sonja

**Review by Coduz — APPROVED** [review]

**Pull request merged**